### PR TITLE
#18 - add files created by SWC python-cookiecutter

### PR DESCRIPTION
**Pull request by niksirbi**
Generated necessary python package files using the Sainsbury Wellcome Center [python-cookiecutter](https://github.com/SainsburyWellcomeCentre/python-cookiecutter/tree/porting_to_pyproject.toml). Will test that the Python package installs properly and all necessary tools work, and then I will open this PR for review

### CODE REVIEW

**Comment by niksirbi** [comment]
As far as I can tell, the package works as intended:
* The package pip installs with: `pip install -e .[dev]`
* The pre-commit hooks are configured to run `isort`, `flake8`, `black`, and `mypy`
* GitHub actions are set up to run linting and testing and to deploy on PyPi if a release is made on GitHub.

The only deviation from the pre-agreed [development guidelines](https://hackmd.io/@i1EOuEktR1OzOa4jaySmKw/rk9UbtIrj) concerns versioning. We have decided to stop using `bump2version`, and instead rely on `setuptools_scm`. I will added specific documentation about that later.

File: README.md [hunk 1]
```diff
@@ -1,4 +1,39 @@
 # WAZP - **W**asp **A**nimal-tracking **Z**oo project with **P**ose estimation
 
-Project name is a subject to refinement.
-CONTRIBUTING.md and pre-commit/testing/package layout will be copied from cookiecutter template.
+Project name is subject to refinement.
+
+Package layout and configuration (including pre-commit hooks and GitHub actions) have been copied from the [python-cookiecutter](https://github.com/SainsburyWellcomeCentre/python-cookiecutter) template.
```
Comment (samcunliffe) [inline README.md:5]:
```suggestion
```

Propose to move this down to its own section 👇

File: README.md [hunk 2]
```diff
@@ -1,4 +1,39 @@
...
+
+```bash
+cd WAZP
+pytest
+mypy -p wazp
+```
+### Versioning and deployment
+The package is deployed to PyPI automatically when a new release is created on GitHub. The version number is automatically determined from the latest tag on the `main` branch.
+
+Details on how this is handled with `setuptools_scm` **TBD**.
```
Comment (samcunliffe) [inline README.md:39]:
```suggestion
Details on how this is handled with `setuptools_scm` **TBD**.

### License

⚖️ [BSD 3-Clause](./LICENSE)

### Template
This package layout and configuration (including pre-commit hooks and GitHub actions) have been copied from the [python-cookiecutter](https://github.com/SainsburyWellcomeCentre/python-cookiecutter) template.
```

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Operating System :: OS Independent",
+    "License :: OSI Approved :: BSD License",
+]
+
+[project.urls]
+homepage = "https://github.com/SainsburyWellcomeCentre/WAZP"
+bug_tracker = "https://github.com/SainsburyWellcomeCentre/WAZP/issues"
+documentation = "https://github.com/SainsburyWellcomeCentre/WAZP"
```
Comment (samcunliffe) [inline pyproject.toml:25]: Comment above about `gh-pages`.

**Review by samcunliffe — APPROVED** [review]
Nice!

**Comment by samcunliffe** [comment]
> 
 The only deviation from the pre-agreed [development guidelines](https://hackmd.io/@i1EOuEktR1OzOa4jaySmKw/rk9UbtIrj) concerns versioning. We have decided to stop using `bump2version`, and instead rely on `setuptools_scm`. I will added specific documentation about that later.

Sorry, I forgot to say: no strong opinion from me.

File: .github/workflows/test_and_deploy.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          python-version: ${{ matrix.python-version }}
+
+  build_sdist_wheels:
+    name: Build source distribution
+    needs: [test]
+    if: github.event_name == 'push' && github.ref_type == 'tag'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: neuroinformatics-unit/actions/build_sdist_wheels@main
+
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply .github/workflows/test_and_deploy.yml:46]: We are already planning to automate docs builds in the cookiecutter. Our current idea is to use Sphinx and automate the build with GitHub Actions (sending the build to a `gh-pages` branch. See for example [this repo](https://github.com/neuroinformatics-unit/troubleshooting), building this [website](https://troubleshooting.neuroinformatics.dev/). That uses the pydata-sphinx-theme.
 Will gladly add to this repo as well. I can open an issue.

[the remark this answers is not in the document]
Comment (samcunliffe) [reply .github/workflows/test_and_deploy.yml:46]: Ah cool. [This is the same "elastic bands and sticky tape"](https://github.com/neuroinformatics-unit/troubleshooting/blob/main/.github/workflows/build_and_publish_docs.yml#L39-L49).

Yeah, I'd say we probably want this eventually.

[the remark this answers is not in the document]
Comment (niksirbi) [reply .github/workflows/test_and_deploy.yml:46]: Since both of us have independently converged on the same 'hacky' solution, let's adopt that for now

**Review by sfmig — APPROVED** [review]
Looks great ✨

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,100 @@
+[project]
+name = "WAZP"
+authors = [{name = "Nikoloz Sirmpilatze", email= "niko.sirbiladze@gmail.com"}]
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply pyproject.toml:3]: Yeap, sorry about that. When creating the repo from the template, one author is passes as an argument. I will add all of us now.

[the remark this answers is not in the document]
Comment (niksirbi) [reply pyproject.toml:3]: @sfmig and @samcunliffe please let me know which email addresses you would like to add here. Should I just use the UCL emails?

[the remark this answers is not in the document]
Comment (sfmig) [reply pyproject.toml:3]: ucl email is fine by me! thanks Niko

[the remark this answers is not in the document]
Comment (samcunliffe) [reply pyproject.toml]: s.cunliffe@ucl.ac.uk, please. But also: it's probably more important that it's you and Sofia. Don't delay merging for putting my email in here.

File: .github/workflows/test_and_deploy.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        # Run all supported Python versions on linux
+        python-version: ["3.8", "3.9", "3.10"]
+        os: [ubuntu-latest]
+        # Include one windows and macos run
+        include:
+        - os: macos-latest
+          python-version: "3.10"
+        - os: windows-latest
+          python-version: "3.10"
+
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply .github/workflows/test_and_deploy.yml:38]: Yes it is

File: pyproject.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (niksirbi) [reply pyproject.toml]: Thank you all. I will merge this as soon as current checks have passed

**Pull request merged**